sklearn/preprocessing/tests/test_function_transformer.py

```diff
@@ -196,9 +195,7 @@ def test_function_transformer_raise_error_with_mixed_dtype(X_type):
     data = _convert_container(data, X_type, columns_name=["value"], dtype=dtype)
 
     def func(X):
-        return np.array(
-            [mapping[_safe_indexing(X, i)] for i in range(X.size)], dtype=object
-        )
+        return np.array([mapping[X[i]] for i in range(X.size)], dtype=object)
```

Comment:
I suspect the `_safe_indexing` is used because `X` can be a pandas series and `X[i]` uses the index and `_safe_indexing` uses the position. For this test, the index happens to be `[0, 1, 2, 3,...,]`, so using `X[i]` works.

For now, I am okay with the work around.